query-algebrizer/src/types.rs

```diff
@@ -334,11 +334,25 @@ pub enum ColumnConstraint {
         left: QueryValue,
         right: QueryValue,
     },
-    HasType(TableAlias, ValueType),
+    HasTypes {
+        value: TableAlias,
+        value_types: ValueTypeSet,
+        strict: bool,
+    },
     NotExists(ComputedTable),
     Matches(QualifiedAlias, QueryValue),
 }
 
+impl ColumnConstraint {
+    pub fn has_type(value: TableAlias, value_type: ValueType) -> ColumnConstraint {
```

Comment:
TODO: expand this to a `ValueTypeSet`.

Reply:
This is intended to be a helper for the most common case, e.g. the single type case where loose checks are desired.